ControlFlowArraysAndStrings/ControlFlowArraysAndStrings/Program.cs

```diff
@@ -0,0 +1,343 @@
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Adds a specified value to the end of the specified array
+        /// and returns the new array.
+        /// </summary>
+        /// <param name="array">The array to be changed.</param>
+        /// <param name="value">The value to be added to the array.</param>
+        /// <returns>The new array with a value added to the last index.</returns>
+        public static string[] AddElementToEnd(string[] array, string value)
+        {
+            string[] newArray = new string[array.Length + 1];
+
+            for (int i = 0; i < array.Length; i++)
+            {
+                newArray[i] = array[i];
+            }
+
+            newArray[newArray.Length - 1] = value;
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Adds a specified value to the specified index of the specified array
+        /// and returns the new array.
+        /// </summary>
+        /// <param name="array">The array to be changed.</param>
+        /// <param name="value">The index where the new value should be added.</param>
+        /// <param name="value">The value to be added to the array.</param>
+        /// <returns>A new array with a value added to the specified index.</returns>
+        public static string[] AddElementAnywhere(string[] array, int index, string value)
+        {
+            string[] newArray = new string[array.Length + 1];
+
+            newArray[index] = value;
+
+            // Copy values from old array into the new array.
+            // Skip over the newly added value.
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (i < index)
+                {
+                    newArray[i] = array[i];
+                }
+
+                if (i > index)
+                {
+                    newArray[i] = array[i - 1];
+                }
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Removes the value at the beginning of a specified array.
+        /// </summary>
+        /// <param name="array">The array to be changed.</param>
+        /// <returns>A new array with a value removed from the start.</returns>
+        public static string[] RemoveElementFromStart(string[] array)
+        {
+            string[] newArray = new string[array.Length - 1];
+
+            for (int i = 1; i < array.Length; i++)
+            {
+                newArray[i - 1] = array[i];
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Removes the value at the end of the specified array.
+        /// </summary>
+        /// <param name="array">The array to be changed.</param>
+        /// <returns>A new array with a value removed from the end.</returns>
+        public static string[] RemoveElementFromEnd(string[] array)
+        {
+            string[] newArray = new string[array.Length - 1];
```

Comment:
You could use your general purpose `RemoveElementFromAnywhere` to implement the specialized `RemoveElementFromEnd`.

Reply:
Wow that suggestion really simplified the `RemoveElementFromEnd`. Thanks!